src/fondant/compiler.py

```diff
@@ -2,23 +2,30 @@
 import json
 import logging
 import typing as t
-from abc import ABC, abstractmethod
+from abc import ABC
 from dataclasses import asdict, dataclass
 from pathlib import Path
 
 import yaml
 
+from fondant.manifest import Metadata
 from fondant.pipeline import Pipeline
 
 logger = logging.getLogger(__name__)
 
 
 class Compiler(ABC):
-    """Abstract base class for a compiler."""
+    """Abstract base class for a pipeline compiler."""
 
-    @abstractmethod
-    def compile(self, *args, **kwargs) -> None:
-        """Abstract method to invoke compilation."""
+    def __init__(
+        self,
+        pipeline: Pipeline,
+    ):
+        self.pipeline = pipeline
```

Comment:
I originally proposed to move the `pipeline` argument to the `compile` method, and I still think that's more logical.

> Why is the pipeline passed as an initialization argument instead of to the `compile` method. From a logical perspective, I wouldn't expect a compiler instance to be specific to a single pipeline.



_Originally posted by @RobbeSneyders in https://github.com/ml6team/fondant/pull/194#discussion_r1227781344_

Reply:
oh I see, makes sense. I'll revert the changes